### DISCUSSION TITLE
How to import pore network for obtaining the permeability

### DISCUSSION THREAD

**Discussion by Soheil1993** [PMEAL/OpenPNM#2401] [Q&A]
Hello everyone,

I hope you are doing well.

I have modeled the compaction of sand particles in Yade and then I have extracted the pore networks (pore and throat data) of each iteration in both txt and vtk format. (It should be mentioned that I did not use PoreSpy and SNOW algorithm for this purpose)

Now I want to obtain the permeability of each of these networks.

Would you please let me know can I use openPNM for this purpose?

I would really appreciate it if you could provide me with further information regarding the importing of my pore networks to openPNM and obtaining the permeability.

I attached the results of one iteration for your kind consideration.

Many thanks in advance.

Best regards,
Soheil

![4 0000](https://user-images.githubusercontent.com/94166206/166231100-a9657f59-da50-4e4e-8397-44b1290ce18b.png)

[yade-network-009000-throats.txt](https://github.com/PMEAL/OpenPNM/files/8602653/yade-network-009000-throats.txt)
[yade-pores-009000.txt](https://github.com/PMEAL/OpenPNM/files/8602654/yade-pores-009000.txt)
[vtk file.zip](https://github.com/PMEAL/OpenPNM/files/8602674/vtk.file.zip)

**Comment by jgostick** [comment]
I don't think that permeability estimation is a good use of PNMs.  You can do basic estimation, but it is a bit of an art to get a good prediction.  PNMs are good for simulating more complex transport and multiphase flow, etc.  Usually you ensure your extraction was good by making sure the PNM's permeability matches a known value. 

You can predict the permeability of an image using LBM which is useful because no effort or thought is required to generate the mesh.  Palabos is a good tool which we have used in the past.

**Reply by Soheil1993** [reply]
Dear Dr. @jgostick 

Thank you very much for your kind reply and valuable information.

You are right. I am at the beginning of my project and I need just a basic estimation of permeability. Also, we considered that the media is fully saturated.

After that I will try to use the LBM model with Palabos.

I could find these examples, but I do not know if I can use them for my case or not.

In this case you used OpenPNM.Utilies.IO.Statoil class to import a network in ‘dat’ format.

(https://openpnm.org/examples/contrib/loading_statoil_finding_permeability.html)

and in these cases you imported the networks from the PoreSpy using SNOW algorithm.

https://github.com/PMEAL/OpenPNM/blob/dev/tests/integration/PoreSpyIO_on_Berea.py

https://openpnm.org/examples/contrib/predicting_effective_permeability_of_berea.html

Would you please let me know how I can import my ready pore network in txt or vtk file and then use these examples?

Best regards,
Soheil

**Reply by PratyushSwain101** [reply]
Mr. Soheil,
were you able to get the way to perform the permeability simulation using PNM?
\

**Reply by Soheil1993** [reply]
Unfortunately no. I am using YADE software and there is a possibility to use DEM–PFV for permeability measurement there.
